platform/source/mbed_retarget.cpp

```diff
@@ -783,6 +922,26 @@ extern "C" ssize_t read(int fildes, void *buf, size_t length)
     }
 }
 
+#if MBED_CONF_PLATFORM_STDIO_MINIMAL_CONSOLE_ONLY
+/* Read the serial interface and store the content to a buffer */
+MBED_WEAK ssize_t minimal_console_read(void *buffer, size_t length)
+{
+    if (length == 0) {
+        return 0;
+    }
+
+    MinimalConsole *mc = get_minimal_console();
+    if (mc == nullptr) {
+        errno = EBADF;
+        return -1;
+    }
+
+    unsigned char *buf = static_cast<unsigned char *>(buffer);
+    buf[0] = mc->getc();
```

Comment:
Check that buffer is not NULL

Reply:
https://github.com/hugueskamba/mbed-os/compare/984d3d625c22b050e42c283372303827d347f0d4..1712d7aa3c8623fa770dcad6516b100ee96f8cf3